image/templates/helm/shared/templates/_scanner-v4_volume.tpl

```diff
@@ -40,7 +40,9 @@
     {{ $_ := include "srox.mergeInto" (list $scannerV4DBPVCCfg $._rox.scannerV4DBPVCDefaults (dict "createClaim" .Release.IsInstall)) }}
   {{ end }}
   {{ $_ = set $scannerV4DBVolumeCfg "persistentVolumeClaim" (dict "claimName" $scannerV4DBPVCCfg.claimName) }}
-  {{ if $scannerV4DBPVCCfg.createClaim }}
+
+  {{ $pvcExists := (lookup "v1" "PersistentVolumeClaim" .Release.Namespace $scannerV4DBPVCCfg.claimName)}}
```

Comment:
Are you aware of the `safeLookup` wrapper for `lookup` that we have in our Helm codebase? I think this wrapper is used in all places where we attempt a lookup -- in addition to that the `.reliable` field of that template's output is checked.

I'm also not sure about any potential negative side-effects this change would have on the operator logic. I think lookups are completely switched off within the operator, using `useLookup: false` or something. This is so to keep the result of applying a Helm chart deterministic and in particular independent of any pre-existing resources. This is probably directly related to the rationale for keeping PVCs managed within the operator itself and not bound to the Helm lifecycle(?).

Reply:
Discussed in a meeting.

Removed the lookup and the pre-hook of the PVC. 

With this change the PVC is managed within the helm lifecycle which automatically involves logic to check if the resources already exists.

I ran the listed manual tests again after this change.